src/app/shared/toggle/collection-favorite-toggle.service.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Lumeer.io, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Injectable} from '@angular/core';
+import {Store} from '@ngrx/store';
+import {AppState} from '../../core/store/app.state';
+import {ToggleService} from './toggle.service';
+import {Collection} from '../../core/store/collections/collection';
+import {CollectionsAction} from '../../core/store/collections/collections.action';
+
+@Injectable()
+export class CollectionFavoriteToggleService extends ToggleService<Collection> {
```

Comment:
Toggling some state is not easy to implement, because user can click 100 times and we don't want to send 100 requests on server. Therefore, I think it's good idea to implement some common behavior for it. It is used in 3 different situations.